src/instruction/set.ts

```diff
@@ -2,25 +2,29 @@ import { Halfword } from 'types/binary'
 import { VirtualBoardError, VirtualBoardErrorType } from 'types/error'
 
 import {
-  IInstruction,
   IInstructionEncoder,
   IInstructionExecutor,
   IInstructionSet
 } from './interfaces'
 import { match } from './opcode'
 
-import { MovInstruction } from './instructions/mov'
+import {
+  MovInstruction,
+  MovsFromRegisterInstruction,
+  MovsFromLiteralInstruction
+} from './instructions/mov'
+import { BaseInstruction } from './instructions/baseInstruction'
 
 export class InstructionSet implements IInstructionSet {
-  private readonly instructions: IInstruction[]
+  private readonly instructions: BaseInstruction[]
```

Comment:
I think this should remain `IInstruction` and not be `BaseInstruction`.

Reply:
or do you mean just the property and not the constructor param?

Reply:
I would not force to use of `BaseInstruction`. The `InstructionSet` class only needs `IInstructionSet`. It does not need to know about the other stuff in the base class.

Reply:
did a fix back to `IInstruction`